File: src/main/java/genetics/Genotype.java
package main;

import Allele;

public class Genotype {
    public String name;

    public Genotype(String name) {
        this.name = name;
    }

    public Allele[] alleles() {
        Allele[] alleles = new Allele[name.length()];
        for(int i = 0; i < alleles.size; i++)
            alleles[i] = name.charAt(i);
        return alleles;
    }

    //traits will eventually be its own class and will account for mutations variations
    public Allele[][] traits() {
        Allele[][] traits = new Allele[2][name.length()/2];
        for(int i = 0; i < alleles().size; i += 2)
            traits[0][i] = alleles()[i];
        for(int i = 1; i < alleles().size; i += 2)
            traits[1][i - 1] = alleles()[i];
        return traits;
    }
}

File: src/main/kotlin/genetics/Genotype.kt
package genetics

import genetics.Trait

import java.util.ArrayList

public class Genotype(var name: String = "__") {
    public var traits: ArrayList<Trait>

    init {
        traits = ArrayList()
        traits.set(0, Trait(name[0], name[1]))
        for (i in 0 until name.length step 2)
            //if(name[i].isLetter())
            traits.add(Trait(name[i], name[i + 1]))
    }

    public fun alleles(): ArrayList<Allele> {
        var alleles = ArrayList<Allele>()
        for(i in 0 until name.length)
            alleles.add(Allele(name[i]))
        return alleles
    }

    public override fun toString(): String {
        return name
    }
}
